tests/src/testcommon.hh

```diff
@@ -278,8 +259,181 @@ template <typename NonLinearOperator, typename FiniteElement>
   return t;
 }
 
+template <typename NonLinearOperator, typename FiniteElement>
+[[nodiscard]] auto checkLinearStressOf3DSimplex(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
+                                                const std::string& messageIfFailed = "") {
+  static_assert(FiniteElement::Traits::mydim == 3,
+                "The test to check linear stress is only supported for the three-dimensional case");
+  assert(fe.localView().element().type() == Dune::GeometryTypes::simplex(3)
+         && "The test to check linear stress is only supported for 3d simplexes (tetrahedrons)");
+  static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == 3,
+                "The test to check linear stress is only supported with the powerBasis being three-dimensional");
+  static_assert(std::is_same_v<std::remove_cvref_t<decltype(fe.localView().tree().child(0))>,
+                               Dune::Functions::LagrangeNode<
+                                   std::remove_cvref_t<decltype(fe.localView().globalBasis().gridView())>, 1, double>>,
+                "The test to check linear stress is only supported for a linear Lagrange basis");
+
+  Dune::TestSuite t("Linear Stress check of 3D solid element (12-node hexahedral)");
+
+  using namespace Ikarus;
+  using namespace Dune::Indices;
+  using namespace Dune::Functions::BasisFactory;
+
+  auto& displacement = nonLinearOperator.firstParameter();
+  displacement << 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0;
+
+  Dune::FieldVector<double, 6> expectedStress
+      = {576.923076900000, 1346.15384610000, 576.923076900000, 0, 384.615384600000, 769.230769200000};
+
+  auto resultRequirements = Ikarus::ResultRequirements<>()
+                                .insertGlobalSolution(Ikarus::FESolutions::displacement, displacement)
+                                .addResultRequest(ResultType::linearStress);
+  auto gridView        = fe.localView().globalBasis().gridView();
+  auto scalarBasis     = makeConstSharedBasis(gridView, lagrangeDG<1>());
+  auto localScalarView = scalarBasis->localView();
+  std::vector<Dune::FieldVector<double, 3>> stressVector(scalarBasis->size());
+
+  ResultTypeMap<double> result;
+
+  auto ele = elements(gridView).begin();
+  localScalarView.bind(*ele);
+  const auto& fe2 = localScalarView.tree().finiteElement();
```

Comment:
Can you check what of these things are actually used? For me it looks like, that from `gridView` up to `fe2` nothing is used below

Reply:
Yeah i think thats unnessesary, i kinda copied it from the test function for Q1 element